sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -635,7 +635,6 @@ object FunctionRegistry {
       (implicit tag: ClassTag[T]): (String, (ExpressionInfo, FunctionBuilder)) = {
     val constructors = tag.runtimeClass.getConstructors
       .filter(_.getParameterTypes.head == classOf[String])
-    assert(constructors.length == 1)
```

Comment:
`assert(constructors.length >= 1)`? cuz these classes must have the name field in the head?

Reply:
`assert(constructors.length >= 1, s"there is no constructor for ${tag.runtimeClass} which takes String as first argument")`

cc @maropu

Reply:
sounds good

Reply:
+1, thanks!